src/network.rs

```diff
@@ -101,11 +102,12 @@ impl Future for TimeoutLimitedBroadcast {
             return Poll::Ready(replace(&mut self.complete, vec![]));
         }
 
-        unsafe {
-            if let Poll::Ready(_) =
-                Pin::new_unchecked(&mut self.timeout).poll(cx)
-            {
-                return Poll::Ready(replace(&mut self.complete, vec![]));
+        if let Some(ref mut timeout) = self.timeout {
+            unsafe {
+                if let Poll::Ready(_) = Pin::new_unchecked(timeout).poll(cx) {
```

Comment:
@stjepang do you see this as dangerous for Pin?

Reply:
No, this looks perfectly fine.